limit RC gossip to 20 peers max

majestrate opened the pull request:
only gossip RCs to 20 peers at a time maximum.
this should reduce the load on service nodes when gossiping RCs.

majestrate committed:
limit RC gossip to 20 peers max

jagerman commented on llarp/router/rc_gossiper.cpp:
```diff
@@ -76,17 +76,38 @@ namespace llarp
     gossip.msgs.emplace_back(new dht::GotRouterMessage(dht::Key_t{}, 0, {rc}, false));
 
-    // send it to everyone
+    constexpr size_t MaxGossipPeers = 20;
```
Should be a bit more prominent, at least in the header for this.

majestrate replied:
moved to header

majestrate committed:
try not to prefer outbound vs inbound link sessions
before when we get the list of router ids for gossip it was highly bias towards outbound sessions.
instead now we get a full list of link session router ids in random order, truncate them to be at most MaxGossipPeers number of keys, and then put them into an unordered set

majestrate committed:
use std::sample to grab MaxGossipPeers when populating std::unordered_set

jagerman committed:
Disable tests on static bionic

It's failing to link with LTO on bionic because of an internal compiler
error for some random reason, so just disable building the tests there
for now (keeping LTO on is more useful since that is also what gets used
for a static build).
